specs/interop/liquidity_migration.md

```diff
@@ -0,0 +1,358 @@
+  - Implement a valid interface, such as `IOptimismMintableERC20`
+    or `IL2StandardToken`.
+  - Deployed with 0 supply: tokens are native to L1,
+    so legacy supply can only grow with deposits from L1.
+  - Can only be minted by the `L2StandardBridge`.
+  - Doesn’t implement fancy functions:
+    e.g. additional mint rights, ability to change l1token, l2bridge, etc.
+    This is hard to measure and will require social consensus.
+- Any `SuperchainERC20` used for conversion will need to allow
+  the `L2StandardBridge` to mint and burn tokens.
+  This can be achieved with an extension to the standard or by overriding
+  the mint and burn functions.
+- For custom implementations: Can we have clusters of allowed L2 representations for the same L1 token?
+  This means that each representation can convert within the same cluster but not with the others,
+  even if the L1 token matches. - Example: CRV in L1 has CRVa and CRVb as two representations in an L2.
+  Can I have a SuperCRVa and a SuperCRVb so that CRVa can only convert to SuperCRVa
+  and CRVb converts to SuperCRVb? - If we do, then we need to know for each Legacy Factory
+  which is the allowed Superchain Factory - If not, it’s enough to compare
+  the L1 token (REMOTE_TOKEN)
+
+## Appendix
+
+### Access Control
+
+Each `SuperchainERC20` will correspond to a a remote token and can be connected
+to multiple existing legacy representations.
+Furthermore, each legacy representation can be connected to various
+`SuperchainERC20` (one per metadata).
+This also implies that users can freely move between allowed legacy
+representations and `SuperchainERC20s` using the convert functions
+(call `convertToSuper(legacy1Addr,...)` and then
+`convertFromSuper(legacy2Addr,...)` for example).
+As legacy tokens can be used to unlock liquidity in the native chain
+(L1 for tokens in the Standard Bridge), it is fundamental only
+to allow safe representations to connect.
+
+Both `convertFromSuper` and `convertToSuper` will first
+check the validity of the input addresses by calling the `checkPair`
+internal function.
+The function will check that the `SuperchainERC20` address can
+be converted with the `OptimismMintableERC20` address.
+
+The `checkPair` function will call a Token List which will act as an allow-list
+of tokens that unify representations that used outdated but legit factories.
+This Token List will consist of mapping legacy addresses to remote addresses so
+it can work as an allow-list and remove the need to check for interface
+differences between `ILegacyMintableERC20`
+and `IOptimismMintableERC20`.
+The Token List will be a one-time update conducted by Optimism and will
+live in the `OptimismMintableERC20Factory` contract.
+If the legacy token is not included in the token list
+(for new tokens, for instance),
+the `checkPair` will verify validity against the factory.
+
+### Decimal management
+
+It is possible to remove the revert on decimal difference
```

Comment:
I lean towards just asserting that the decimals are the same and not worrying about scaling them

Reply:
Yes, that seems to be the consensus. Would you remove the whole section?